test/jsupdate-behaviour-mocha.js

```diff
@@ -0,0 +1,156 @@
+// jsupdater-behaviour-mocha.js
+
+var chai = require('chai');
+chai.should();
+chai.use(require('chai-as-promised'));
+var sinon = require('sinon');
+
+var noflo = require('noflo');
+var test = require('./common-test');
+var jswrapper = require('../src/javascript-wrapper');
+
+/**
+ * Theses test are intented to verify the state management of the system.
+ * @see https://github.com/rdf-pipeline/noflo-rdf-pipeline/issues/35
+ */
+describe("framework-state", function() {
+    it("should fire updater when it has valid input states on all of its attached inputs", function() {
+        return new Promise(function(done, fail){
+            return test.createNetwork({
+                node1: "core/Repeat",
+                node2: "core/Repeat",
+                sut: jswrapper(function(input1, input2) {
+                    return "Hello " + input1 + " and " + input2;
+                })
+            }).then(function(network){
+                network.graph.addEdge('node1', 'out', 'sut', 'input1');
+                network.graph.addEdge('node2', 'out', 'sut', 'input2');
+                test.onOutPortData(network.processes.sut.component, 'error', fail);
+                test.onOutPortData(network.processes.sut.component, 'output', done);
+                test.sendData(network.processes.node1.component, 'in', "from node1");
+                test.sendData(network.processes.node2.component, 'in', "from node2");
+            });
+        }).should.eventually.have.property('data', "Hello from node1 and from node2");
+    });
+    it("should not fire updater if not all of its attached inputs have valid states", function() {
+        return new Promise(function(done, fail){
+            return test.createNetwork({
+                node1: "core/Repeat",
+                node2: "core/Repeat",
+                sut: jswrapper(function(input1, input2) {
+                    fail("Hello " + input1 + " and " + input2);
+                })
+            }).then(function(network){
+                network.graph.addEdge('node1', 'out', 'sut', 'input1');
+                network.graph.addEdge('node2', 'out', 'sut', 'input2');
+                test.sendData(network.processes.node1.component, 'in', "from node1");
+                setTimeout(done.bind(this, "nothing happened"), 100);
+            });
+        }).should.become("nothing happened");
+    });
+    it("should not fire updater if upstream updater did not produce an output state", function() {
+        return new Promise(function(done, fail){
+            return test.createNetwork({
+                sink: jswrapper(function(input) {
+                    // do nothing
```

Comment:
Note: This looks right, but I think that the framework will have to be modified to handle this case.